Sources/ApolloCodegenLib/CodegenConfiguration/ApolloCodegenConfiguration+OperationManifestConfiguration.swift

```diff
@@ -0,0 +1,179 @@
+    public struct Default {
+      public static let operationManifest: OperationManifestFileOutput? = nil
+      public static let operationDocumentFormat: OperationDocumentFormat = .definition
+      public static let autoGenerate: Bool = false
+    }
+    
+    // MARK: - Initializers
+    
+    /// Designated initializer
+    ///
+    /// - Parameters:
+    ///   - operationManifes: The `OperationManifestFileOutput` used to determine where and how to output the operation manifest JSON
+    ///   - operationDocumentFormat: The `OperationDocumentFormat` used to determine how to output operations in the generated code files.
+    public init(
+      operationManifest: OperationManifestFileOutput? = Default.operationManifest,
+      operationDocumentFormat: OperationDocumentFormat = Default.operationDocumentFormat,
+      autoGenerate: Bool = Default.autoGenerate
+    ) {
+      self.operationManifest = operationManifest
+      self.operationDocumentFormat = operationDocumentFormat
+      self.autoGenerate = autoGenerate
+    }
+    
+    // MARK: - Codable
+    
+    enum CodingKeys: CodingKey, CaseIterable {
+      case operationManifest
+      case operationDocumentFormat
+      case autoGenerate
+    }
+    
+    public init(from decoder: Decoder) throws {
+      let values = try decoder.container(keyedBy: CodingKeys.self)
+      try throwIfContainsUnexpectedKey(
+        container: values,
+        type: Self.self,
+        decoder: decoder
+      )
+      
+      operationManifest = try values.decode(
+        OperationManifestFileOutput.self,
+        forKey: .operationManifest
+      )
+      
+      operationDocumentFormat = try values.decode(
+        OperationDocumentFormat.self,
+        forKey: .operationDocumentFormat
+      )
+      
+      autoGenerate = try values.decode(
+        Bool.self,
+        forKey: .autoGenerate
+      )
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+      var container = encoder.container(keyedBy: CodingKeys.self)
+      
+      try container.encode(self.operationManifest, forKey: .operationManifest)
+      try container.encode(self.operationDocumentFormat, forKey: .operationDocumentFormat)
+      try container.encode(self.autoGenerate, forKey: .autoGenerate)
+    }
+    
+    // MARK: - OperationManifestFileOutput
+    
+    /// Configures the generation of an operation manifest JSON file for use with persisted queries
+    /// or [Automatic Persisted Queries (APQs)](https://www.apollographql.com/docs/apollo-server/performance/apq).
+    ///
+    /// The operation manifest is a JSON file that maps all generated GraphQL operations to an
+    /// operation identifier. This manifest can be used to register operations with a server utilizing
+    /// persisted queries
+    /// or [Automatic Persisted Queries (APQs)](https://www.apollographql.com/docs/apollo-server/performance/apq).
+    /// Defaults to `nil`.
+    public struct OperationManifestFileOutput: Codable, Equatable {
```

Comment:
When we move `operationDocumentFormat` back into it's original spot, we can probably just have `path` and `version` be properties on the top level `OperationManifestConfiguration` along with `autoGenerate`.